sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -277,13 +291,25 @@ class CodegenContext {
       funcName: String,
       funcCode: String,
       inlineToOuterClass: Boolean = false): String = {
+    val newFunction = addNewFunctionInternal(funcName, funcCode, inlineToOuterClass)
+    newFunction match {
+      case NewFunction(functionName, None, None) => functionName
+      case NewFunction(functionName, Some(_), Some(subclassInstance)) =>
+        subclassInstance + "." + functionName
+    }
+  }
+
+  private[this] def addNewFunctionInternal(
+      funcName: String,
+      funcCode: String,
+      inlineToOuterClass: Boolean): NewFunction = {
     // The number of named constants that can exist in the class is limited by the Constant Pool
     // limit, 65,536. We cannot know how many constants will be inserted for a class, so we use a
-    // threshold of 1600k bytes to determine when a function should be inlined to a private, nested
+    // threshold of 1000k bytes to determine when a function should be inlined to a private, nested
```

Comment:
Any reason to change this?

Reply:
Is it for the possibly to be add method grouping all split methods? If yes, we can add the method into an inner class (see https://github.com/apache/spark/pull/19480#discussion_r144240879), so we don't need to change this threshold.

Reply:
yes, without this change, I got the Constant Pool limit exception of the {{NestedClass}}. Actually I tried to address this issue in this PR (https://github.com/apache/spark/pull/19447), but without the other changes in thee current PR I wasn't able to create a UT for it.

Reply:
Can you try https://github.com/apache/spark/pull/19480#discussion_r144240879, so we may avoid changing this threshold?

Reply:
I can't change that for the reason I explained there.